leaf/extensions/tags.py

```diff
@@ -65,26 +65,32 @@ async def tag_autocomplete(
         ]
 
     @app_commands.describe(
+        user="The user you wish to search for.",
         starting_page="The page to start on.",
         silent="Whether the response should only be visible to you.",
     )
     @app_commands.command(name="list", description="Lists all the tags in the server.")
     async def list_tags(
         self,
         interaction: discord.Interaction,
+        user: Optional[discord.Member] = None,
         starting_page: Optional[int] = 1,
         silent: Optional[bool] = False,
     ) -> None:
-        tags = await self.bot.database.fetch(
-            "SELECT * FROM tags WHERE guild_id = $1 AND deleted = FALSE",
-            interaction.guild.id,
-        )
-        embeds = []
+        if user is not None:
+            query = "SELECT * FROM tags WHERE guild_id = $1 AND owner_id = $2 AND deleted = FALSE"
+            tags = await self.bot.database.fetch(query, interaction.guild.id, user.id)
+        else:
+            query = "SELECT * FROM tags WHERE guild_id = $1 AND deleted = FALSE"
+            tags = await self.bot.database.fetch(query, interaction.guild.id)
 
-        if not tags:
+        embeds = []
+        if len(tags) == 0:
```

Comment:
Why not `if not tags` or `if not len(tags)`?

Reply:
Ill update that.